create-new-data-pr.py

```diff
@@ -89,6 +109,10 @@ def update_tag_version(current_version=None):
           print('files were modified, update mid version and reset minor', sec, thrd)
       new_tag = 'V'+first+'-'+sec+'-'+thrd
       # message should be referencing the PR that triggers this job
+      pr_branch_sha = data_repo.get_branch(data_pr_base_branch).commit.sha
+      #there is a bug in old versions, we need this three calls instead of one: https://github.com/PyGithub/PyGithub/issues/1336
+      gh_tag = data_repo.create_git_tag(tag=new_tag,message='Details in: '+data_repo_pr.html_url,object=pr_branch_sha,type='tag type')
```

Comment:
@mrodozov , just creating the tag is enough. There is no need to create ref and release

Reply:
there is a bug, see the comment and the URL.
In the version we use create_git_tag it doesn't create the tag, I checked it. I took the example that worked from that github issue

Reply:
ah ok, so create_git_tag only create a tag object ... do we need to create this tag objec? I think only calling
```
data_repo.create_git_ref(ref='refs/tags/'+new_tag, sha=pr_branch_sha)
```
is enough to create a tag

Reply:
yes we need it's gh_tag.sha which is used to create the tag as a reference in 
`create_git_ref(ref='refs/tags/'+new_tag, sha=gh_tag.sha)`. so we only create it to get the tag sha from it which is later used to create the actual tag , meaning the  create_git_tag is only half working but just enough to make use of it.
The release is ... for human reading :D I can remove the release if we don't want it

Reply:
gh_tag.sha is same as pr_branch_sha so there is no need to create an tag object

Reply:
it's not, this was my first thought the tag has it's own sha, not the one that it's tagging.
I'm checking it again but I'm sure (because I already did)

Reply:
On my repo https://github.com/smuzaffar/RecoBTag-Combined , the CMSSW_10_6_X branch sha is `aab813614cc883665f5ac86ff01c9a8e09741caa` ( https://api.github.com/repos/smuzaffar/RecoBTag-Combined/branches ) and the following works for me 
```
data_repo.create_git_ref(ref='refs/tags/X00-03-01', sha='aab813614cc883665f5ac86ff01c9a8e09741caa')
```

Reply:
this is the output of this test
```
print("gh_tag.sha is:", gh_tag.sha, " pr_branch_sha is: ", pr_branch_sha) 
gh_tag.sha is: 929412f85052dd98168a85fe216ab2fb57d02243  pr_branch_sha is:  b473a61a7611abbadd45aa23fcc2dc21da1cc23d
```
the gh_tag.sha is the tag object sha, the pr_branch_sha is different

Reply:
this is what I get
```
>>> data_repo.get_branch('CMSSW_10_6_X').commit.sha
u'aab813614cc883665f5ac86ff01c9a8e09741caa'
```
so just using `data_repo.get_branch('CMSSW_10_6_X').commit.sha` as sha is good enogh to create the ref ( why make an extra API call? )

Reply:
we only need lightweight tags 

https://docs.github.com/en/rest/reference/git#tags
Note that creating a tag object does not create the reference that makes a tag in Git. If you want to create an annotated tag in Git, you have to do this call to create the tag object, and then create the refs/tags/[tag] reference. **If you want to create a lightweight tag, you only have to create the tag reference - this call would be unnecessary.**

Reply:
I'm trying it

Reply:
it works you are right, only creating a reference with the pr commit tags it properly.

Reply:
do we want the release or you want me to remove the call as well (as discussed being redundant)
sorry, I didn't notice this:
https://github.com/cms-sw/cms-bot/pull/1480/files#r569306705
in time

Reply:
deleyte the release creation call. For data repo we only need tag

Reply:
it's deleted